lib/modules/manager/devcontainer/extract.ts

```diff
@@ -0,0 +1,95 @@
+import { logger } from '../../../logger';
+import { isValidDependency } from '../custom/regex/utils';
+import { getDep as getDockerDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { DevContainerFile } from './schema';
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+  extractConfig: ExtractConfig,
+): PackageFileContent | null {
+  try {
+    const file = DevContainerFile.parse(content);
+    const deps: PackageDependency[] = [];
+
+    const image = file?.image ?? null;
+    const imageDep = getDep(image, packageFile, extractConfig.registryAliases);
+
+    if (imageDep) {
+      deps.push(imageDep);
+    } else {
+      logger.debug(
+        { packageFile },
+        'No image defined in dev container JSON file.',
+      );
+    }
+
+    const features = file.features ?? {};
+
+    if (features) {
```

Comment:
remove, it's always true if you use the `?? {}` above. you can also remove the `?? {}`

Reply:
Ive removed the `?? {}`